### PR TITLE
Block Locking: Styling changes for modal

### PR DESCRIPTION
**Pull request by Mamaduka**
## What?
PR addresses general block locking modal feedback from https://github.com/WordPress/gutenberg/pull/39950#issuecomment-1084917076.

Changes made in this PR:

* Removes the separator lines
* Increase padding so nested items line up correctly.
* Swap the existing icons for locked/unlocked icon indicators.
* Add hover state.
* Use [a new method](https://github.com/WordPress/gutenberg/pull/39462/) for Checkbox indeterminate state. This will fix how the `minus` icon is displayed.

## Testing Instructions
1. Open a Post or Page
2. Insert a Heading Block.
3. Open lock modal from Block options.
4. Confirm updates work as expected.

## Screenshots or screencast 
CleanShot 2022-04-03 at 15 46 16

### CODE REVIEW

**Comment by paaljoachim** [comment]
Great!

### Before:
Screenshot 2022-04-03 at 17 32 13

### After:
Screenshot 2022-04-03 at 17 34 56

What I miss here is the lines. As the lines have a way of creating a stronger connection between text and icon. One can easier see what is locked and what is unlocked. (Instead of adding the invisible line in the mind one can see the line visually.)
Lock-modul-adding-back-lines

@critterverse

**Comment by jasmussen** [comment]
Nice work! I like the absence of borders, we use them sparingly in other places as well.

Two things: if possible we should change this CSS at the component level, i.e. affect the same interface that exists in the Preferences modal, so we don't need to have CSS that overwrites:

Screenshot 2022-04-04 at 08 14 58

If we did that, it would look like this:

Screenshot 2022-04-04 at 08 16 50

I'll entirely defer to @critterverse on getting these details right, she has some context that I don't have. One idea to submit is that if the icons on the right are a concern, we could move them left of the label, like so:
Screenshot 2022-04-04 at 08 18 11

That would also give some connective tissue to other places that list blocks:

Screenshot 2022-04-04 at 08 18 24

What do you think?

**Comment by Mamaduka** [comment]
> if possible we should change this CSS at the component level, i.e. affect the same interface that exists in the Preferences modal, so we don't need to have CSS that overwrites

We don’t have separate components to build similar UIs, so not sure if this is possible.

But it would be nice to revisit them in the future. There might a be new experimental components we could use.

**Comment by jasmussen** [comment]
Cool, I'm happy to give a thumbs up, but I'd love a check by Channing!

**Review by jasmussen — APPROVED** [review]

**Comment by Mamaduka** [comment]
Thank you, @paaljoachim and @jasmussen. Let's wait for @critterverse's approval :)

**Review by critterverse — APPROVED** [review]

**Comment by critterverse** [comment]
This looks so much cleaner, thanks @Mamaduka!!

**Pull request merged**
